streams/src/test/java/org/apache/kafka/streams/processor/internals/GlobalStateManagerImplTest.java

```diff
@@ -0,0 +1,406 @@
+        stateDirectory = new StateDirectory("appId", stateDirPath);
+        consumer = new MockConsumer<>(OffsetResetStrategy.EARLIEST);
+        stateManager = new GlobalStateManagerImpl(topology, consumer, stateDirectory);
+        checkpointFile = new File(stateManager.baseDir(), ProcessorStateManager.CHECKPOINT_FILE_NAME);
+    }
+
+    @After
+    public void after() throws IOException {
+        stateDirectory.unlockGlobalState();
+    }
+
+    @Test
+    public void shouldLockGlobalStateDirectory() throws Exception {
+        stateManager.initialize(context);
+        assertTrue(new File(stateDirectory.globalStateDir(), ".lock").exists());
+    }
+
+    @Test(expected = StreamsException.class)
+    public void shouldThrowStreamsExceptionIfCantGetLock() throws Exception {
+        final StateDirectory stateDir = new StateDirectory("appId", stateDirPath);
+        try {
+            stateDir.lockGlobalState(1);
+            stateManager.initialize(context);
+        } finally {
+            stateDir.unlockGlobalState();
+        }
+    }
+
+    @Test
+    public void shouldReadCheckpointOffsets() throws Exception {
+        final Map<TopicPartition, Long> expected = writeCheckpoint();
+
+        stateManager.initialize(context);
+        final Map<TopicPartition, Long> offsets = stateManager.checkpointedOffsets();
+        assertEquals(expected, offsets);
+    }
+
+    @Test
+    public void shouldDeleteCheckpointFileAfteLoaded() throws Exception {
+        writeCheckpoint();
+        stateManager.initialize(context);
+        assertFalse(checkpointFile.exists());
+    }
+
+    @Test(expected = StreamsException.class)
+    public void shouldThrowStreamsExceptionIfFailedToReadCheckpointedOffsets() throws Exception {
+        final File checkpointFile = new File(stateManager.baseDir(), ProcessorStateManager.CHECKPOINT_FILE_NAME);
+        try (final FileOutputStream stream = new FileOutputStream(checkpointFile)) {
+            stream.write("0\n1\nblah".getBytes());
+        }
+        stateManager.initialize(context);
+    }
+
+    @Test
+    public void shouldInitializeStateStores() throws Exception {
+        stateManager.initialize(context);
+        assertTrue(store1.initialized);
+        assertTrue(store2.initialized);
+    }
+
+    @Test
+    public void shouldReturnInitializedStoreNames() throws Exception {
+        final Set<String> storeNames = stateManager.initialize(context);
+        assertEquals(Utils.mkSet(store1.name(), store2.name()), storeNames);
+    }
+
+    @Test
+    public void shouldThrowIllegalArgumentIfTryingToRegisterStoreThatIsNotGlobal() throws Exception {
+        stateManager.initialize(context);
+
+        try {
+            stateManager.register(new NoOpReadOnlyStore<>("not-in-topology"), false, new TheStateRestoreCallback());
+            fail("should have raised an illegal argument exception as store is not in the topology");
+        } catch (final IllegalArgumentException e) {
+            // pass
```

Comment:
No, i disagree. Most tests i'd use `@Test(expected = ....)` in would either have a single method call in them, so you know that is the only method that can throw the exception. Or, it would be a `new Blah(..)` followed by testing the single method.  Of course the `new ..` could throw an exception, but hopefully most ctors are side-effect free.